src/main/java/com/lpnu/shaggybeavers/auth/AuthService.java

```diff
@@ -0,0 +1,59 @@
+package com.lpnu.shaggybeavers.auth;
+
+import com.lpnu.shaggybeavers.model.User;
+import com.lpnu.shaggybeavers.repository.CategoryRepository;
+import com.lpnu.shaggybeavers.repository.UserRepository;
+import com.lpnu.shaggybeavers.security.UserPrincipal;
+import com.lpnu.shaggybeavers.security.jwt.JwtUtil;
+import com.lpnu.shaggybeavers.service.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final UserRepository userRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+    private final JwtUtil jwtUtil;
+    private final AuthenticationManager authenticationManager;
+
+    public AuthResponse register(RegisterRequest request) {
```

Comment:
We need discus about that, about returning type and you musn't inject repository use service layer